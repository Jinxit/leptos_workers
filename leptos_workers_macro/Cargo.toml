[package]
name = "leptos_workers_macro"
version = "0.1.0"
edition = "2021"
description = "Macros for leptos_workers, a WebWorker library for the Leptos web framework."
license = "MIT OR Apache-2.0"

[lib]
name = "leptos_workers_macro"
path = "src/lib.rs"
proc-macro = true

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
proc-macro2 = "1.0"
quote = "1.0"
syn = { version = "2.0", features = ["full"] }
proc-macro-error = "1.0"