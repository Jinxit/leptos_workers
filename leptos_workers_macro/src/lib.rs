#![warn(clippy::pedantic)]
#![allow(clippy::unnecessary_wraps)]
#![allow(clippy::too_many_arguments)]

mod v2;

use crate::v2::analyze::analyze;
use crate::v2::codegen::codegen;
use crate::v2::lower::lower;
use crate::v2::parse::parse;
use proc_macro_error::proc_macro_error;

extern crate proc_macro2;

#[proc_macro_error]
#[proc_macro_attribute]
pub fn worker(
    args: proc_macro::TokenStream,
    item: proc_macro::TokenStream,
) -> proc_macro::TokenStream {
    let ast = parse(&args.into(), item.into());
    let model = analyze(ast);
    let ir = lower(&model);
    let rust = codegen(ir);
    rust.into()
}
