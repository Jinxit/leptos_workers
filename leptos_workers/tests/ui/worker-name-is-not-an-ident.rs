use leptos_workers_macro::worker;

#[worker(struct)]
async fn future_worker(req: TestRequest) -> TestResponse {}

fn main() {}
