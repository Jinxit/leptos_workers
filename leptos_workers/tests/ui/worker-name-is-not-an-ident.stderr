error: invalid worker name: `struct`
 --> tests/ui/worker-name-is-not-an-ident.rs:3:10
  |
3 | #[worker(struct)]
  |          ^^^^^^
